inference-engine/src/multi_device/multi_device_async_infer_request.cpp

```diff
@@ -47,7 +48,9 @@ MultiDeviceAsyncInferRequest::MultiDeviceAsyncInferRequest(
                                _multiDeviceExecutableNetwork->_devicePrioritiesInitial.cbegin(),
                                _multiDeviceExecutableNetwork->_devicePrioritiesInitial.cend(),
                                [&name](const MultiDevicePlugin::DeviceInformation& d){
-                                    return d.deviceName == name; });
+                                    DeviceIDParser deviceParser(name);
+                                    return d.deviceName == name || d.deviceName == deviceParser.getDeviceName();
```

Comment:
probably, we should compare `r->getDeviceName()` with `deviceName + execNetwork.GetConfig(CONFIG_KEY(DEVICE_ID)).as<string>()`

In this case we can match exactly devices

Reply:
Fixed